driver/src/driver/stablex_driver.rs

```diff
@@ -41,14 +41,21 @@ impl<'a> StableXDriver<'a> {
             solution
         };
 
-        let submitted = if solution.executed_sell_amounts.iter().sum::<u128>() > 0 {
-            self.contract.submit_solution(batch, orders, solution)?;
-            info!("Successfully applied solution to batch {}", batch);
-            true
+        let current_objective_value = self.contract.get_current_objective_value()?;
+        let submitted = if let Some(objective_value) = solution.objective_value {
+            if current_objective_value < objective_value {
+                self.contract.submit_solution(batch, orders, solution)?;
+                info!("Successfully applied solution to batch {}", batch);
+                true
+            } else {
+                info!("Not submitting trivial solution for batch {}", batch);
```

Comment:
This log message seems off.

Reply:
I agree. Also, this wouldn't be possible anyway.